### PR TITLE
Added new NER metric using fuzzywuzzy logic

### PR DESCRIPTION
**Pull request by sarathsgvr**
Added a new metric using fuzzywuzzy logic for NER evaluation. This will help to compare the predictions with actual using string match based on predefined threshold.

### CODE REVIEW

File: tests/library/test_metrics.py [hunk 1]
```diff
@@ -1432,3 +1433,35 @@ def test_llm_as_judge_metric(self):
...
+                ("Marathahalli ring road", "Location"),
+            ]
+        ]
+    ]
+    outputs = apply_metric(
+        metric=metric, predictions=predictions, references=references
+    )
+    global_target = 1.0
+    self.assertAlmostEqual(global_target, outputs[0]["score"]["global"]["f1_Person"])
+    global_target = 0.0
```
Comment (yoavkatz) [inline tests/library/test_metrics.py:1460]: Would Ner() metric return a different result?

File: prepare/metrics/customf1fuzzy.py [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+from unitxt.test_utils.metrics import test_metric
+
+metric = FuzzyNer()
+# Test1 single line single class
+# 0.1 simple case, multi examples
+predictions = [
+    [("Amir", "Person"), ("Yaron", "Person")],
+    [("Ran", "Person"), ("Yonatan", "Person")],
+]
+references = [[[("Yaron", "Person"), ("Ran", "Person")]], [[("Yonatan", "Person")]]]
```
Comment (yoavkatz) [inline prepare/metrics/customf1fuzzy.py:12]: I think we need to modify the tests to see that there Fuzzy matching makes a difference compared to regular NER.

File: tests/library/test_metrics.py (hunk 1 above)
Comment (rrajkumar1990) [reply tests/library/test_metrics.py:1460]: we have changed the predicton and reference inputs

File: prepare/metrics/customf1fuzzy.py (hunk 2 above)
Comment (rrajkumar1990) [reply prepare/metrics/customf1fuzzy.py]: we will update different data to check on the values

File: src/unitxt/metrics.py [hunk 3]
```diff
@@ -3642,3 +3644,118 @@ class NormalizedSacrebleu(HuggingfaceMetric):
...
+                        max_score = score
+                        best_total_key = total_key
+
+            if best_total_key is not None:
+                tmp.append(min(actual_group[actual_key], total_group[best_total_key]))
+            else:
+                tmp.append(min(actual_group[actual_key], 0))
+        return sum(tmp), sum(actual_group.values())
+
+    def compute(
```
[the remark this answers is not in the document]
Comment (rrajkumar1990) [reply src/unitxt/metrics.py]: Renamed calculate_groups_ratio_fuzzy to calculate_groups_ratio. And removed the compute method  accordingly to use the same base compute method

File: tests/library/test_metrics.py (hunk 1 above)
Comment (yoavkatz) [reply tests/library/test_metrics.py:1460]: But NER would match "jar htaras" to "jar htaras" and not match  "Marathahalli" "Marathahalli ring road", exactly as the fuzzy match does.   I would expect the test to match   "Marathahalli" and "Marathahalli ring road".

File: src/unitxt/metrics.py [hunk 4]
```diff
@@ -12,6 +13,7 @@
 import evaluate
 import numpy
 import numpy as np
+from fuzzywuzzy import fuzz
```
Comment (yoavkatz) [inline src/unitxt/metrics.py:16]: Please import this where used and not in main scope - so if people don't use the metric, it will not fail.

**Comment by rrajkumar1990** [comment]
> Hi. I think primary what's left is to update the test value, and move the import to where it is used. Is this right?

We have change the Test values and changed the import inside the custom method.

**Comment by yoavkatz** [comment]
> > Hi. I think primary what's left is to update the test value, and move the import to where it is used. Is this right?
> 
> We have changed the Test values and changed the import inside the custom method.

I saw you moved the import.  But I don't see changes in the test values. Am I missing anything?

File: tests/library/test_metrics.py (hunk 1 above)
Comment (yoavkatz) [reply tests/library/test_metrics.py]: Tests added.

**Pull request merged**